db/models.py

```diff
@@ -50,3 +59,60 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class User(AbstractUser):
+    pass
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        to=User, on_delete=models.CASCADE, related_name="orders"
+    )
+
+    def __str__(self) -> str:
+        return f"{self.created_at}"
+
+    class Meta:
+        ordering = ("-created_at",)
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        to=MovieSession, on_delete=models.CASCADE, related_name="tickets"
+    )
+    order = models.ForeignKey(
+        to=Order, on_delete=models.CASCADE, related_name="tickets"
+    )
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    def __str__(self) -> str:
+        return (
+            f"{self.movie_session.movie.title} {self.movie_session.show_time} "
+            f"(row: {self.row}, seat: {self.seat})"
+        )
+
+    def clean(self) -> None:
+        cinema_hall = self.movie_session.cinema_hall
+        if not (1 <= self.row <= cinema_hall.rows):
+            raise ValidationError({
+                "row": f"row number must be in available range: "
+                f"(1, rows): (1, {cinema_hall.rows})"
+            })
+        if not (1 <= self.seat <= cinema_hall.seats_in_row):
+            raise ValidationError({
+                "seat": f"seat number must be in available range: "
+                f"(1, seats_in_row): (1, {cinema_hall.seats_in_row})"
+            })
+
+    def save(self, *args, **kwargs) -> Callable:
+        self.full_clean()
+        return super().save(*args, **kwargs)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=["movie_session", "row", "seat"],
+                                    name="unique_ticket"),
```

Comment:
constraint name should contain names of fields that are unique + Meta class should be moved above str method